### PR TITLE
Add GET /api/v1/follow_requests Mastodon API endpoint

### PR DESCRIPTION
**Pull request by MrPetovan**
Part of #7887 

This is only the GET endpoint but it's a good start to show how the future Mastodon endpoints will look.

### CODE REVIEW

**Comment by nupplaphil** [comment]
We should have an eye on the tests as well. You introduced a new route for API requests. Is there a possibility to check the API similar to the `include/ApiTest.php` tests? With the existing data of the dataset fixture? At least I'd prefere if you add a `src/Module/Api/Mastodon/FollowTest.php` file with some tests you WOULD do if you would have the right data for it (like `testFollowRequestWithUnknownUser()` or `testFollowRequestForUser()`, ...) so we cann add test-sets at a "later stage" ( ;-) )

**Comment by MrPetovan** [comment]
It was easier to test the API when they weren't modules, but now I'm not sure how to proceed since we haven't added tests for modules yet.

**Comment by nupplaphil** [comment]
Yes you're right .. the problem is the `header()` and `System::jsonExit()` method, because they aren't "overwritable" so there's currently no chance to get the result of the module call.

Just add empty test-methods what you think what is reasonable to test. I think we should talk about how we use the "header" and "..Exit" methods.. 
- Encapsulate the "header" call inside a/the "..Exit" method
- Moving System::jsonExit to the "base" API call => so depending on the format (xml, atom, rss, ..) you call the right "...Exit" based on the output of the module call. So we could call the module out of the base-api context and check the return ;-)

**Comment by annando** [comment]
@nupplaphil whatever we do - we should do it in a way that doesn't complicate coding. I would prefer keeping the static methods but being able to inject something to them. Possibly the static methods could simply be wrapper for dynamic ones.

I suggest creating a new class only for API calls. There we could define the functions that then could be hijacked for testing.

Something different: Could you add some documentation about the new API calls? I would suggest a new document. And I think we don't need to describe each and every parameter, but we should refer to the Mastodon definition and should only document differences here.

**Comment by MrPetovan** [comment]
A different document than API.md?

**Comment by annando** [comment]
I think we should separate that. There we could also document all currently not supported API endpoints.

**Comment by MrPetovan** [comment]
I started rewriting the API help, separating the current Twitter and GNU Social endpoints, adding a third sub-page for Mastodon endpoints.

**Comment by annando** [comment]
The old API uses `api_user()` instead of `local_user()`, see here: https://github.com/friendica/friendica/blob/develop/include/api.php#L69-L71

Do you think that we should keep that?

**Comment by MrPetovan** [comment]
> Do you think that we should keep that?

Why not? `api_user()` uses `local_user()` anyway?

**Comment by annando** [comment]
`api_user` uses `local_user` but additionally checks an additional session variable to ensure that the person really logged in via API.

**Comment by MrPetovan** [comment]
I'm not sure what's your point? What would you change/remove/replace?

**Comment by annando** [comment]
My point is this check here: https://github.com/friendica/friendica/pull/7904/files#diff-101b8007a9a50e9c86484b546c5aab4dR46-R48

This had to be changed to the `api_user` - and later surely some API class function.

**Comment by MrPetovan** [comment]
This was lifted directly from `api_post()` in `mod/api.php`.

**Comment by annando** [comment]
Then I consider it as a bug there as well.

**Comment by MrPetovan** [comment]
I can fix it.

**Comment by annando** [comment]
Could you add a "user" function to the API base class that is (currently) simply calling the "api_user" function? I would like to accept this PR, especially since I would like to add some endpoints here that are widely used for statistics stuff (that wouldn't publish more data like the existing ones - but on more well known API endpoints)

**Comment by MrPetovan** [comment]
That's what I inadvertently did with `getCurrentUserId` which is ill-named. But I'm rewriting the whole API docs at the moment so the PR isn't ready.

**Comment by annando** [comment]
Ah okay. Good to know. Then I will wait until you finished your job.

**Comment by MrPetovan** [comment]
This PR has been updated with @annando 's feedback but not @nupplaphil because I'm stumbling on the duality JSON/XML. The former could use a returned object as is, but the latter requires a top-level element name, and as a result I have no idea how to make the API module return just the data because it wouldn't work with the XML outputs.

**Comment by annando** [comment]
Good work so far!

Where exactly is the problem? BTW: I don't think that we need XML for the Mastodon endpoints.

**Comment by nupplaphil** [comment]
I think we can refactor this part at a later stage as well.. But we should really create the RC branch to have a nice christmas present for @tobiasd ;-)

File: doc/API-Entities.md [hunk 1]
```diff
@@ -0,0 +1,1458 @@
+# Friendica API entities
+
+* [Home](help)
+  * [Using the APIs](help/api)
+
+
+## Activities
+
+<table class="table table-condensed table-striped table-bordered">
```
Comment (nupplaphil) [inline doc/API-Entities.md]: Why didn't up you use the markdown specific way of creating tables? https://help.github.com/en/github/writing-on-github/organizing-information-with-tables I'd prefer it over the plain HTML :)

File: doc/API-Friendica.md [hunk 2]
```diff
@@ -0,0 +1,682 @@
...
+Returns the API user's [Photo List Items](help/API-Entities#Photo+List+Item).
+
+#### Return values
+
+json:
+
+```json
+	[
+		{
+			id: "resource_id",
```
Comment (nupplaphil) [inline doc/API-Friendica.md]: Isn't the key of a json a string too?

File: doc/API-Entities.md (hunk 1 above)
Comment (MrPetovan) [reply doc/API-Entities.md]: No way of styling the generated tables.

File: doc/API-Friendica.md (hunk 2 above)
Comment (MrPetovan) [reply doc/API-Friendica.md]: It actually showed up as an error in my IDE!

**Comment by MrPetovan** [comment]
> Where exactly is the problem?

The goal is for API methods to be agnostic of the return format, and returning only the data structure. But XML needs an additional piece of information in the form of the top-level element. So we can't just return the list of follow requests. A solution would be to add a superseding return structure like

```json
{
    "root": "follow_requests",
    "data": [
        {
            "id": ...
        },
        ...
    ]
}
```

And have the JSON formatter only return the `data` part.

**Comment by MrPetovan** [comment]
Can we please merge this before I start adding the POST endpoints?

**Comment by annando** [comment]
I waited for your signal to merge it :-) Do you want to wait for a reply from @nupplaphil?

**Comment by nupplaphil** [comment]
Before writing "approved", I merged it ;)

**Comment by annando** [comment]
@MrPetovan BTW: I would like to add some statistics endpoints as well. Do you think that this is possible - or would we interfere too much?

**Comment by MrPetovan** [comment]
What do you mean? How would it interfere with my work?

**Comment by annando** [comment]
We would both work in the API segment.

**Comment by MrPetovan** [comment]
It's fine, I started having more than one file for API endpoints. You can either fiddle with `include/api.php` or have your own Module class in `src/` and we won't interfere with each other.

**Pull request merged**